packages/flutter/lib/src/material/segmented_button.dart

```diff
@@ -436,6 +444,16 @@ class SegmentedButton<T> extends StatefulWidget {
 @visibleForTesting
 class SegmentedButtonState<T> extends State<SegmentedButton<T>> {
   bool get _enabled => widget.onSelectionChanged != null;
+  bool _hovering = false;
+  bool _focused = false;
+  bool get _selected => widget.selected.isNotEmpty;
+
+  Set<WidgetState> get _states => <WidgetState>{
```

Comment:
When we disable one of the segments by setting `ButtonSegment.enabled` to false, `_states` cannot really add `WidgetState.disabled` to the set, so the `disabledBorder` in line 628 is actually not the real disabled border and then incorrect side is applied to the disabled segment. This causes the google testing failures.